### ISSUE TITLE
Unable to list CSP subscription

### ISSUE THREAD

**Issue by shahabhijeet** [Azure/azure-powershell#4085]
List-AzureRmAccount -TenantId <tenantId>
List-AzureRmSubscription -TenantId <tenantId>

Does not show list of subscription associated with the tenant.
These subscriptions are CSP subscription.
So this issue is for CSP subscriptions

Works fine with AzureRm.Module 2.4.0
Regression in AzureRm.Module 3.0.0

## Debug Output

> Get-AzureRmSubscription -TenantId ed713048-b431-4635-a717-714a81cad80a -Debug

```
DEBUG: 3:04:42 PM - GetAzureRMSubscriptionCommand begin processing with ParameterSet 'ListByIdInTenant'.

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): A
DEBUG: 3:04:44 PM - using account id 'DanRoyal@testazurecspautomation.onmicrosoft.com'...
DEBUG: [Common.Authentication]: Authenticating using configuration values: Domain: 'Common', Endpoint:
'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirect:
'urn:ietf:wg:oauth:2.0:oob', ResourceClientUri: 'https://management.core.windows.net/', ValidateAuthrity: 'True'
DEBUG: [Common.Authentication]: Acquiring token using context with Authority
'https://login.microsoftonline.com/Common/', CorrelationId: '00000000-0000-0000-0000-000000000000', ValidateAuthority:
'True'
DEBUG: [Common.Authentication]: Acquiring token using AdalConfiguration with Domain: 'Common', AdEndpoint:
'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirectUri:
urn:ietf:wg:oauth:2.0:oob
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 06/06/2017 22:04:44: 9cd53831-0dc4-4ae6-9826-37757ad0620b - AcquireTokenHandlerBase: === Token Acquisition
started:
 Authority: https://login.microsoftonline.com/Common/
 Resource: https://management.core.windows.net/
 ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
 CacheType: Microsoft.Azure.Commands.Common.Authentication.ProtectedFileTokenCache (15 items)
 Authentication Target: User

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 06/06/2017 22:04:44:  - TokenCache: Deserialized 15 items to token cache.
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 06/06/2017 22:04:44: 9cd53831-0dc4-4ae6-9826-37757ad0620b - TokenCache: Looking up cache for a token...
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 06/06/2017 22:04:44: 9cd53831-0dc4-4ae6-9826-37757ad0620b - TokenCache: No matching token was found in the cache
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 06/06/2017 22:04:44:  - WindowsFormsWebAuthenticationDialogBase: Navigating to
'https://login.microsoftonline.com/Common/oauth2/authorize?resource=https://management.core.windows.net/&client_id=1950
a258-227b-4e31-a9cf-717495945fc2&response_type=code&haschrome=1&redirect_uri=urn:ietf:wg:oauth:2.0:oob&login_hint=DanRo
yal@testazurecspautomation.onmicrosoft.com&client-request-id=9cd53831-0dc4-4ae6-9826-37757ad0620b&prompt=attempt_none&x
-client-SKU=.NET&x-client-Ver=2.28.3.860&x-client-CPU=x64&x-client-OS=Microsoft Windows NT
10.0.15063.0&site_id=501358&display=popup'.
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 06/06/2017 22:04:44:  - WindowsFormsWebAuthenticationDialogBase: Navigating to
'https://login.microsoftonline.com/Common/oauth2/authorize?resource=https://management.core.windows.net/&client_id=1950
a258-227b-4e31-a9cf-717495945fc2&response_type=code&haschrome=1&redirect_uri=urn:ietf:wg:oauth:2.0:oob&login_hint=DanRo
yal@testazurecspautomation.onmicrosoft.com&client-request-id=9cd53831-0dc4-4ae6-9826-37757ad0620b&prompt=attempt_none&x
-client-SKU=.NET&x-client-Ver=2.28.3.860&x-client-CPU=x64&x-client-OS=Microsoft Windows NT
10.0.15063.0&site_id=501358&display=popup&sso_nonce=AQABAAAAAABnfiG-mA6NTae7CdWW7Qfd3wIirk3-ZTsKGHuUxkZIeyfo71YhztVvC1P
sOF1Jnsd63JghsrOZn2hYewP2LJaXlYUlKpv3cCpiWHYRHmR89yAA&mscrid=9cd53831-0dc4-4ae6-9826-37757ad0620b'.
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 06/06/2017 22:04:44: 9cd53831-0dc4-4ae6-9826-37757ad0620b - TokenCache: Storing token in the cache...
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 06/06/2017 22:04:44: 9cd53831-0dc4-4ae6-9826-37757ad0620b - TokenCache: An item was stored in the cache
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 06/06/2017 22:04:44: 9cd53831-0dc4-4ae6-9826-37757ad0620b - AcquireTokenHandlerBase: === Token Acquisition
finished successfully. An access token was retuned:
 Access Token Hash: eBHL226D7ymPIu+q0Zzq004TQ0WTkg6dD17nJGQvJ+g=
 Refresh Token Hash: oSbOFEL4Rb44L02Ky4WsJSW4qWal26cHhQ/XLUzRpJ8=
 Expiration Time: 06/06/2017 23:04:43 +00:00
 User Hash: PxlmSwgmE5uoqDKICPEISVLqeCdc1cNs9mssjwBMMwk=

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 06/06/2017 22:04:44:  - TokenCache: Serializing token cache with 15 items.
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://management.azure.com/tenants?api-version=2016-06-01

Headers:
x-ms-client-request-id        : 7b1f056c-c2b3-49af-9177-1044141b2a90
accept-language               : en-US

Body:


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
Pragma                        : no-cache
x-ms-ratelimit-remaining-tenant-reads: 14999
x-ms-request-id               : 1d03dd1e-1872-4bd9-b451-19596fa1023d
x-ms-correlation-request-id   : 1d03dd1e-1872-4bd9-b451-19596fa1023d
x-ms-routing-request-id       : WESTUS:20170606T220445Z:1d03dd1e-1872-4bd9-b451-19596fa1023d
Strict-Transport-Security     : max-age=31536000; includeSubDomains
Cache-Control                 : no-cache
Date                          : Tue, 06 Jun 2017 22:04:44 GMT

Body:
{
  "value": [
    {
      "id": "/tenants/04081b9f-c8b4-424b-86b9-ce44721c2b23",
      "tenantId": "04081b9f-c8b4-424b-86b9-ce44721c2b23"
    }
  ]
}

DEBUG: 3:04:45 PM - GetAzureRMSubscriptionCommand end processing.
DEBUG: 3:04:45 PM - GetAzureRMSubscriptionCommand end processing.
```

**Comment by cormacpayne** [comment]
Need to add the following code before [this line](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs#L251):

```cs
if (!string.IsNullOrWhiteSpace(tenant))
{
    return new List<AzureTenant>() { CreateTenant(tenant) };
}
```

**Comment by cormacpayne** [comment]
Fix is in PR https://github.com/Azure/azure-powershell/pull/4087

**Comment by cormacpayne** [comment]
Fix is now available in 4.1.0
